cnf-tests/testsuites/e2esuite/security/tuning_sysctl.go

```diff
@@ -0,0 +1,179 @@
+			podDefinition := pods.DefineWithNetworks(TestNamespace, []string{fmt.Sprintf("%s/%s", namespaces.SRIOVOperator, "test-network")})
+			pod, err := client.Client.Pods(TestNamespace).Create(context.Background(), podDefinition, metav1.CreateOptions{})
+			Expect(err).ToNot(HaveOccurred())
+			err = pods.WaitForCondition(client.Client, pod, corev1.ContainersReady, corev1.ConditionTrue, 1*time.Minute)
+			Expect(err).ToNot(HaveOccurred())
+		})
+
+		It("pods with sysctl's on bond over sriov interfaces should start", func() {
+			// NOTE: due to a bond cni bug we need to specify the name of the bond interface in both the cni config and
+			// the multus annotation for the network
+			bondLinkName := "bond0"
+			err := createBondNetworkAttachmentDefinitionWithSysctls(apiclient, TestNamespace, "bond", bondLinkName, "net1", "net2", map[string]string{Sysctl: "1"})
+			Expect(err).ToNot(HaveOccurred())
+			podDefinition := pods.DefineWithNetworks(TestNamespace, []string{
+				fmt.Sprintf("%s/%s", namespaces.SRIOVOperator, "test-network"),
+				fmt.Sprintf("%s/%s", namespaces.SRIOVOperator, "test-network"),
+				fmt.Sprintf("%s/%s@%s", TestNamespace, "bond", bondLinkName),
+			})
+			pod, err := client.Client.Pods(TestNamespace).Create(context.Background(), podDefinition, metav1.CreateOptions{})
+			Expect(err).ToNot(HaveOccurred())
+			err = pods.WaitForCondition(client.Client, pod, corev1.ContainersReady, corev1.ConditionTrue, 1*time.Minute)
+			Expect(err).ToNot(HaveOccurred())
+		})
+
+	})
+	Context("clean SRIOV configuration", func() {
+		It("should restore the cluster to the original state", func() {
+			if !discovery.Enabled() {
+				By("cleaning the sriov test configuration")
+				networks.CleanSriov(sriovclient, TestNamespace)
+			}
+			err := namespaces.Delete(TestNamespace, apiclient)
+			Expect(err).ToNot(HaveOccurred())
+		})
+	})
+})
+
+func ceateMacVlanNetworkAttachmentDefinition(cs *client.ClientSet, namespace, nadName, ip string, sysctls map[string]string) error {
+	macvlanConfig := fmt.Sprintf(`{"type": "macvlan","ipam":{"type":"static","addresses":[{"address":"%s/24"}]}}`, ip)
+	return createNetworkAttachmentDefinitionWithSysctls(cs, namespace, nadName, macvlanConfig, sysctls)
+}
+
+func createBondNetworkAttachmentDefinitionWithSysctls(cs *client.ClientSet, namespace, nadName, bondName, link1, link2 string, sysctls map[string]string) error {
+	bondConfig := fmt.Sprintf(`{
+	    "type": "bond",
+		"ifname": "%s",
+		"mode": "active-backup",
+		"failOverMac": 1,
+		"linksInContainer": true,
+		"miimon": "100",
+		"links": [ {"name": "%s"}, {"name": "%s"} ],
+        "ipam": {
+            "type": "host-local",
+            "subnet": "1.1.1.0/24"
+        }}`, bondName, link1, link2)
+	return networks.CreateMultiConfigNetworkAttachmentDefinition(cs, namespace, nadName, bondConfig, sysctlConfig(sysctls))
+}
+
+func sysctlConfig(sysctls map[string]string) string {
+	sysctlString := ""
```

Comment:
Isn't the result a json? If so, consider using json.Marshal

Reply:
It's a string all the way until we create a network-attachment-definition

Reply:
Yes, what I am asking here is, are you building a json string manually? If so, I'd use go rendering